src/components/metadata/metadata-code.test.js

```diff
@@ -49,57 +48,3 @@ describe('MetaDataCode', () => {
     ).toBe(true);
   });
 });
-
-describe('MetaCodeToggle', () => {
```

Comment:
I wonder why are these tests deleted? The should not be deleted given the change to use the toggle component, but rather to update the `wrapper` selector below to the new css class for the toggle component

Reply:
Yes, i was a bit confused regarding this bit. So now, there is a toggle component with its own unit tests which are very similar to the MetaCodeToggle tests. Since this is unit testing and the toggle tests are handled by the core component (Toggle.js). I thought maybe it should not be repeated here...

Reply:
I see - I had a look at the tests at `toggle.test.js`, and the tests that are set up there are testing different things - those tests are scoped to just within the toggle component,  while all the tests here are focused on the performance of the code toggle and all relevant actions for the code panel, which is essential to test the entire code panel feature. These should not be deleted - please put them back in. Thank you :)

Reply:
@studioswong , i added the MetaDataToggle tests back and i renamed the component to Toggle and updated the classnames and prop names. Now it looks exactly like the test functions in toggle . So I am not sure -- whenever you are free, can we do a quick 1:1 to address it. thanks. I don't want to push it as I am still not convinced !

Reply:
I had a further detailed look at the tests that were deleted and see that it was set up in a similar fashion to the tests in `toggle.test.js`, and definitely understand where you are coming from. 

For me the main value I am looking for in setting up toggle related tests in `meta-data-code` is to test the `onChange` callback that was passed into the toggle, with which this needs to be set up in the context with the `meta-data-code` file given that the callback was setup and passed on from here ( refer to the `'onChange callback fires when input changed'` test that was deleted here) - the tests set up in `Toggle.js` was scoped to testing the dummy component itself, while setting up the test here would allow us to test it end to end. ( the situation is similar to the difference between tests for individual selectors, while stlll setting up tests within the component that consumes that selector to test the end to end funcionality of that feature). 

I would say the simpliest way to do this is to set up a test similar to the `'onChange callback fires when input changed'` test with the new toggle, while the other tests that only tests the passed prop for the toggle component can stay within `toggle.test.js`. 

Other than the current setup, I would even extend the tests within `toggle.test.js` to ensure that the right css classes gets shown for the different prop ( current it just tests whether the passed prop is right which is pretty much useless, as it's testing an assumed react capability rather than testing our app setup). I know the previous testing setup was really not good and your work was built on from there, so definitely this is no fault on you - let's use this as an oppourtunity to do it well ;) we can pair on this tomorrow in our 1:1.